common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -3801,6 +3799,466 @@ static jint NativeCrypto_EVP_AEAD_CTX_open_buf(JNIEnv* env, jclass, jlong evpAea
                            inBuffer, aadArray, EVP_AEAD_CTX_open);
 }
 
+static jbyteArray NativeCrypto_EVP_HPKE_CTX_export(JNIEnv* env, jclass, jobject hpkeCtxRef,
+                                                   jbyteArray exporterCtxArray, jint length) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    EVP_HPKE_CTX* hpkeCtx = fromContextObject<EVP_HPKE_CTX>(env, hpkeCtxRef);
+    JNI_TRACE("EVP_HPKE_CTX_export(%p, %p, %d)", hpkeCtx, exporterCtxArray, length);
+
+    if (hpkeCtx == nullptr) {
+        // NullPointerException thrown while calling fromContextObject
+        return {};
+    }
+
+    jbyte* exporterCtxArrayElement = nullptr;
+    const uint8_t* exporterCtx = nullptr;
+    size_t exporterCtxLen = 0;
+    if (exporterCtxArray != nullptr) {
+        exporterCtxArrayElement = env->GetByteArrayElements(exporterCtxArray, 0);
+        exporterCtx = reinterpret_cast<const uint8_t*>(exporterCtxArrayElement);
+        exporterCtxLen = env->GetArrayLength(exporterCtxArray);
+    }
+
+    size_t exportedLen = length;
+    std::vector<uint8_t> exported(exportedLen);
+
+    bool isExported = EVP_HPKE_CTX_export(/* ctx= */ hpkeCtx,
+                                          /* out= */ exported.data(),
+                                          /* secret_len= */ exportedLen,
+                                          /* context= */ exporterCtx,
+                                          /* context_len= */ exporterCtxLen);
+
+    if (exporterCtxArrayElement != nullptr) {
+        env->ReleaseByteArrayElements(exporterCtxArray, exporterCtxArrayElement, JNI_ABORT);
```

Comment:
Any reason this isn't using `ScopedByteArrayRO`? That'll be less error-prone, and then you don't need `isExported`, you can just write `if (!EVP_HPKE_CTX_export(...))`

Ditto for each of the other functions.

Reply:
Done. I have updated it to use  `ScopedByteArrayRO` instead.